test/21_fleet-target-state.ts

```diff
@@ -0,0 +1,251 @@
+import * as _ from 'lodash';
+import * as mockery from 'mockery';
+import { expect } from 'chai';
+import { supertest } from './test-lib/supertest';
+import { version } from './test-lib/versions';
+import * as configMock from '../src/lib/config';
+import * as fixtures from './test-lib/fixtures';
+import { Application, User } from '../src/balena-model';
+
+const POLL_MSEC = 2000;
+const TIMEOUT_SEC = 1;
+
+// @ts-expect-error mock the value for the default poll interval...
+configMock['DEFAULT_SUPERVISOR_POLL_INTERVAL'] = POLL_MSEC;
+
+// @ts-expect-error mock the value for the timeout grace period...
+configMock['API_HEARTBEAT_STATE_TIMEOUT_SECONDS'] = TIMEOUT_SEC;
+
+// register the mocks...
+mockery.registerMock('../src/lib/config', configMock);
+
+// test fleet default state helper
+const appMatcherFunc = (svc: AnyObject, fxProp: AnyObject) =>
+	svc.hasOwnProperty('environment') && !!fxProp;
+const svcMatcherFunc = (svc: AnyObject, fxProp: AnyObject) =>
+	svc.id === fxProp.service.__id;
+const imgMatcherFunc = (svc: AnyObject, fxProp: AnyObject) =>
+	svc.image_id === fxProp.release_image.image.__id;
+
+const checkServiceProperties = (
+	fixture: AnyObject,
+	fleetStateReleasesServices: AnyObject,
+	propertyCollectionName: string,
+	propKeyIdentifier: string,
+	matcherFunction: (svc: AnyObject, fxProperty: AnyObject) => boolean,
+) => {
+	for (const fxProperty of Object.values(fixture)) {
+		for (const svc of Object.values<any>(fleetStateReleasesServices)) {
```

Comment:
Why the `any`?

Reply:
Yes thanks, leaving it without the template type as any is inferred from the AnyObject type.